config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/BundleEntityBuilder.java

```diff
@@ -32,59 +41,147 @@
-                                                            Document document, String projectName,
-                                                            String projectGroupName, String projectVersion) {
+    public Pair<Element, BundleMetadata> build(Bundle bundle, EntityBuilder.BundleType bundleType,
+                                               Document document, String projectName,
+                                               String projectGroupName, String projectVersion, AnnotatedEntity annotatedEntity) {
         List<Entity> entities = new ArrayList<>();
         entityBuilders.forEach(builder -> entities.addAll(builder.build(bundle, bundleType, document)));
+        if(annotatedEntity != null){
+            if (EntityBuilder.BundleType.DEPLOYMENT == bundleType) {
+
+                List<Entity> entityList = getEntityDependencies(annotatedEntity.getPolicyName(), entities, bundle);
+                LOGGER.log(Level.FINE, "Annotated entity list : " + entityList);
+                Optional<Entity> rootEntity = entityList.stream().filter(entity -> annotatedEntity.getEntityName().equals(entity.getName())).findFirst();
+                if(!rootEntity.isPresent()){
+                    Optional<Entity> foundEntity = entities.stream().filter(entity -> annotatedEntity.getEntityName().equals(entity.getName())).findFirst();
+                    entityList.add(foundEntity.get());
+                }
+                List<Entity> bundleEntities = renameNonReusableEntities(entityList, bundle, annotatedEntity, projectGroupName, projectVersion);
+                // Create bundle and its metadata
+                final Element annotatedBundle = bundleDocumentBuilder.build(document, bundleEntities);
+                final BundleMetadata bundleMetadata = bundleMetadataBuilder.build(annotatedEntity,
+                        bundleEntities, projectGroupName, projectVersion);
+
+                return ImmutablePair.of(annotatedBundle,
+                        bundleMetadata);
+            }
+            return null;
+        } else {
+            return ImmutablePair.of(bundleDocumentBuilder.build(document,
+                    entities), null);
+        }
+    }
+
+    private List<Entity> renameNonReusableEntities(List<Entity> entityList, Bundle bundle, AnnotatedEntity annotatedEntity, String projectGroupName, String projectVersion) {
+        List<Entity> bundleEntities = new ArrayList<>();
+        final Map<String, Policy> policyMap = bundle.getPolicies();
+        final Map<String, Encass> encassMap = bundle.getEncasses();
+        for(Entity entity : entityList){
+            if(EntityTypes.ENCAPSULATED_ASSERTION_TYPE.equals(entity.getType()) || EntityTypes.POLICY_TYPE.equals(entity.getType())) {
+                Encass encassEntity = encassMap.get(entity.getName());
+                Policy policyEntity = policyMap.get(entity.getName());
```

Comment:
Have you tested the scenario where encass and policy has same name.

Reply:
in case of policy its a complete path where as encass will have only name so no issues